pkg/standalone-utils/test/GaugeActions.test.ts

```diff
@@ -0,0 +1,659 @@
+        expect(await lpToken.balanceOf(gauge.address)).to.be.eq(0);
+        expect(await lpToken.balanceOf(tokenSenderAddress)).to.be.eq(totalLpTokens);
+        if (tokenSenderAddress != tokenRecipientAddress) {
+          expect(await lpToken.balanceOf(tokenRecipientAddress)).to.be.eq(0);
+        }
+
+        expect(await gauge.balanceOf(gauge.address)).to.be.eq(0);
+        expect(await gauge.balanceOf(tokenSenderAddress)).to.be.eq(0);
+        expect(await gauge.balanceOf(tokenRecipientAddress)).to.be.eq(0);
+
+        const tx = await relayer.connect(userSender).multicall([
+          encodeDeposit({
+            gauge: gauge,
+            sender: tokenSenderAddress,
+            recipient: tokenRecipientAddress,
+            amount: depositAmountOrRef,
+          }),
+        ]);
+        receipt = await tx.wait();
+      });
+
+      // Short-circuit when no tokens are deposited.
+      expectedAmount.gt(0) &&
+        it('pulls BPT tokens from sender if necessary', async () => {
+          // This transfer is skipped if token sender is relayer, but addresses are undefined outside this scope.
+          if (tokenSenderAddress != relayer.address) {
+            expectTransferEvent(
+              receipt,
+              { from: tokenSenderAddress, to: relayer.address, value: expectedAmount },
+              lpToken.address
+            );
+          }
+        });
+
+      it("approves gauge to use relayer's BPT funds", async () => {
+        expectEvent.inIndirectReceipt(receipt, lpToken.interface, 'Approval', {
+          owner: relayer.address,
+          spender: gauge.address,
+          value: expectedAmount,
+        });
+      });
+
+      // Short-circuit when no tokens are deposited.
+      expectedAmount.gt(0) &&
+        it('emits BPT transfer event from relayer to gauge', async () => {
+          expectTransferEvent(
+            receipt,
+            { from: relayer.address, to: gauge.address, value: expectedAmount },
+            lpToken.address
+          );
+        });
+
+      it('transfers BPT tokens to gauge', async () => {
+        expect(await lpToken.balanceOf(tokenSenderAddress)).to.be.almostEqual(totalLpTokens.sub(expectedAmount));
+        expect(await lpToken.balanceOf(gauge.address)).to.be.eq(expectedAmount);
+      });
+
+      it('emits deposit event', async () => {
+        expectEvent.inIndirectReceipt(receipt, gauge.interface, 'Deposit', {
+          provider: tokenRecipientAddress,
+          value: expectedAmount,
+        });
+      });
+
+      it('mints gauge tokens to recipient', async () => {
+        expect(await gauge.balanceOf(tokenRecipientAddress)).to.be.eq(expectedAmount);
+      });
+
+      it('emits transfer event for minted gauge tokens', async () => {
+        expectTransferEvent(
+          receipt,
+          { from: ZERO_ADDRESS, to: tokenRecipientAddress, value: expectedAmount },
+          gauge.address
+        );
+      });
+    }
+  });
+
+  describe('gaugeWithdraw', () => {
```

Comment:
Most (all?) of the comments on deposit also apply here

Reply:
comments applied to deposit as well.